modules/raft/src/integrationTest/java/org/apache/ignite/raft/jraft/core/ItNodeTest.java

```diff
@@ -3506,6 +3507,102 @@ public void onRawConfigurationCommitted(ConfigurationEntry conf) {
         );
     }
 
+    @Test
+    public void testIndexAndTermOfCfgArePropagatedToSnapshotMeta() throws Exception {
+        TestPeer peer0 = new TestPeer(testInfo, TestUtils.INIT_PORT);
+
+        AtomicLong term = new AtomicLong(-1);
+        AtomicLong index = new AtomicLong(-1);
+
+        AtomicLong metaTerm = new AtomicLong(-1);
+        AtomicLong metaIndex = new AtomicLong(-1);
+
+        cluster = new TestCluster(
+                "testIndexAndTermOfCfgArePropagatedToSnapshotMeta",
+                dataPath,
+                Collections.singletonList(peer0),
+                new LinkedHashSet<>(),
+                ELECTION_TIMEOUT_MILLIS,
+                (peerId, opts) -> {
+                    opts.setFsm(new MockStateMachine(peerId) {
+                        @Override
+                        public boolean onSnapshotLoad(SnapshotReader reader) {
+                            SnapshotMeta meta = reader.load();
+
+                            metaTerm.set(meta.cfgTerm());
+                            metaIndex.set(meta.cfgIndex());
```

Comment:
Can any of 3 nodes write its index+term to atomics or just the leader? How about restricting this explicitly by `peerId`?

Reply:
If we only need indexes and terms from the leader, then why do we need other nodes? Why one of them is stopped?

Reply:
Snapshot load is triggered only on the node, that is restarted during the test.
Other nodes needed to change configuration using `changePeersAndLearnersAsync`, and to force snapshot load on the restarting node

Reply:
My suggestion is to make this (that just one node can change index and term) obvious. An `if` that I proposed in the first comment would solve this. Another option is to add a comment, for example

Reply:
added if